### DISCUSSION TITLE
Homebrew ShellGPT Formula

### DISCUSSION THREAD

**Discussion by vanyauhalin** [TheR1D/shell_gpt#286] [Ideas]
Hey, everyone. I write the formula for myself, but if you're a fan of installing everything through Homebrew, then I'm eager [to share the tap](https://github.com/vanyauhalin/homebrew-shell_gpt) with you.

```sh
$ brew tap vanyauhalin/shell_gpt
$ brew install shell_gpt
```

<p align="center">
  <img width="682" alt="output" src="https://github.com/TheR1D/shell_gpt/assets/56481109/68951757-c3ee-4730-8d2e-5bac3d4405aa">
</p>

Update from Aug 31, 2024:

> [!WARNING]
> This formula is not being updated anymore. The reason is that it has become difficult to reproduce the exact dependency tree required for the Homebrew formula. I thought about distributing ShellGPT as a precompiled binary, but that took a toll on the app's performance. As a result, I switched to another AI terminal tool.

**Comment by carlsmedstad** [comment]
Before finding this I also created [my own formula](https://github.com/carlsmedstad/homebrew-carlsmedstad/blob/master/shell-gpt.rb). What's you opinion on adding an official formula or linking to a community-maintained one in the README, @TheR1D?

**Reply by TheR1D** [reply]
Hi @carlsmedstad, at this time, I'm focusing on supporting PyPI as the primary distribution channel, mainly due to maintenance considerations. While I'm all for community involvement and support community-maintained packages, I'm hesitant to include them in the README. This is primarily due to potential security concerns, as it's challenging to ensure the security and reliability of externally maintained packages.

**Comment by sshakil** [comment]
`pip3 install --user shell-gp` and `pip3 install --user shell-gpt` give `error: externally-managed-environment` with a note on `--break-system-packages` that says "You can override this, at the risk of breaking your Python installation or OS, by..."

grateful for what's here and dist thru brew would be nice
